python/metatensor-torch/metatensor/torch/atomistic/outputs.py

```diff
@@ -172,3 +176,96 @@ def _check_energy(
+
+    if request.per_atom:
+        expected_samples_names = ["system", "atom"]
+    else:
+        expected_samples_names = ["system"]
+
+    if energy_ensemble_block.samples.names != expected_samples_names:
+        raise ValueError(
+            "invalid sample names for 'energy_ensemble' output: "
+            f"expected {expected_samples_names}, got "
+            f"{energy_ensemble_block.samples.names}"
+        )
+
+    # check samples values from systems & selected_atoms
+    if request.per_atom:
+        expected_values: List[List[int]] = []
+        for s, system in enumerate(systems):
+            for a in range(len(system)):
+                expected_values.append([s, a])
+
+        expected_samples = Labels(
+            ["system", "atom"], torch.tensor(expected_values, device=device)
+        )
+        if selected_atoms is not None:
+            expected_samples = expected_samples.intersection(selected_atoms)
+
+        if len(expected_samples.union(energy_ensemble_block.samples)) != len(
+            expected_samples
+        ):
+            raise ValueError(
+                "invalid samples entries for 'energy_ensemble_block' output, they "
+                "do not match the `systems` and `selected_atoms`. "
+                f"Expected samples:\n{expected_samples}"
+            )
+
+    else:
+        expected_samples = Labels(
+            "system", torch.arange(len(systems), device=device).reshape(-1, 1)
+        )
+        if selected_atoms is not None:
+            selected_systems = Labels(
+                "system", torch.unique(selected_atoms.column("system")).reshape(-1, 1)
+            )
+            expected_samples = expected_samples.intersection(selected_systems)
+
+        if len(expected_samples.union(energy_ensemble_block.samples)) != len(
+            expected_samples
+        ):
+            raise ValueError(
+                "invalid samples entries for 'energy_ensemble' output, they do not "
+                "match the `systems` and `selected_atoms`. "
+                f"Expected samples:\n{expected_samples}"
+            )
+
+    if len(energy_ensemble_block.components) != 0:
+        raise ValueError(
+            "invalid components for 'energy_ensemble' output: components "
+            "should be empty"
+        )
+
+    if energy_ensemble_block.properties != Labels(
+        "ensemble_member",
+        torch.arange(energy_ensemble_block.values.shape[1], device=device).reshape(
+            -1, 1
+        ),
+    ):
+        raise ValueError(
+            "invalid properties for 'energy_ensemble' output: expected a Labels "
+            "object with the name 'ensemble_member' and the values being the indices "
+            "of the ensemble members from 0 to the number of ensemble members minus "
+            "one."
+        )
+
+    if len(energy_ensemble_block.gradients()) != 0:
+        raise ValueError(
+            "invalid gradients for 'energy' output: no gradients should be present."
```

Comment:
```suggestion
            "invalid gradients for 'energy_ensemble' output: no gradients should be present."
```

Reply:
Do we actually want to disallow the possibility to have forward gradients for `energy_ensemble`?

Reply:
Yeah... they make no sense... unless we give them a different meaning.
For example, the uncertainties on the forces are not the derivatives of the uncertainties on the energies

Reply:
I mean, the output is not called "energy_uncertainties" but "energy_ensemble", so the positions gradients would be an ensemble of forces. It would then be up to the engine to do what they want with it.

I'm also fine disallowing it for now, but then we should mention it in the standard.

Reply:
You're right, I got confused. The gradients for `energy_ensemble` are well-defined and it's a good idea to allow them as explicit gradients